File: Quivi.Backoffice.Api/Requests/MenuItems/PatchMenuItemRequest.cs
using Quivi.Domain.Entities;
using Quivi.Domain.Entities.Pos;
using Quivi.Infrastructure.Apis;

namespace Quivi.Backoffice.Api.Requests.MenuItems
{
    public class PatchMenuItemRequest
    {
        public string? Name { get; init; }
        public Optional<string> Description { get; init; }
        public Optional<string> ImageUrl { get; init; }
        public decimal? Price { get; init; }
        public PriceType? PriceType { get; init; }
        public decimal? VatRate { get; init; }
        public Optional<string> LocationId { get; init; }
        public IDictionary<Language, PatchMenuItemLanguage>? Translations { get; init; }
        public IEnumerable<string>? MenuCategoryIds { get; init; }
        public IEnumerable<string>? ModifierGroupIds { get; init; }
    }

    public class PatchMenuItemLanguage
    {
        public string? Name { get; init; }
        public Optional<string> Description { get; init; }
    }
}

File: quivi.backoffice.react.webapp/src/hooks/api/Dtos/menuItems/PatchMenuItemRequest.ts
import { Language } from "../Language";
import { PriceType } from "./PriceType";

export interface PatchMenuItemRequest {
    readonly id: string;
    readonly name?: string;
    readonly description?: string | null;
    readonly imageUrl?: string | null;
    readonly price?: number;
    readonly priceType?: PriceType;
    readonly vatRate?: number;
    readonly locationId?: string | null;
    readonly translations?: Record<Language, PatchMenuItemTranslation | undefined>;
    readonly menuCategoryIds?: string[];
    readonly modifierGroupIds?: string[];
}

export interface PatchMenuItemTranslation {
    readonly name?: string;
    readonly description?: string;
}